source/AI.cpp

```diff
@@ -1609,7 +1667,9 @@ bool AI::ShouldDock(const Ship &ship, const Ship &parent, bool playerShipsLaunch
 	
 	// A fighter should retreat if its parent is calling it back, or it is
 	// a player's ship and is not in the current system.
-	if(!parent.Commands().Has(Command::DEPLOY) || (ship.IsYours() && !playerShipsLaunch))
+	if(!parent.Commands().Has(Command::DEPLOY) && !ship.IsYours())
+		return true;
+	if (ship.IsYours() && (!ship.IsDeploying() || ship.GetSystem() != playerSystem))
 		return true;
```

Comment:
Can this logic be combined into an if-else if branch?

Maybe
```c++
if(ship.IsYours())
{
	if(!ship.IsDeploying() || ship.GetSystem() != playerSystem)
		return true;
}
else if(!parent.Commands().Has(Command::DEPLOY))
	return true;
```

At minimum, i think moving the `IsYours` check first for both lines helps clarify the logic that applies to the player ship vs NPC ship cases. Feel free to update the comment above this block as appropriate.

(Also, an `if (` snuck in.)

Reply:
Yes, that looks quite readable; implemented as suggested and updated the comment block(s).